bindings/postgres/postgres.go

```diff
@@ -67,6 +73,14 @@ func (p *Postgres) Init(ctx context.Context, meta bindings.Metadata) error {
 	}
 
 	poolConfig, err := m.GetPgxPoolConfig()
+
+	// Register uuid and decimal types with pgx
+	poolConfig.AfterConnect = func(ctx context.Context, conn *pgx.Conn) error {
+		pgxuuid.Register(conn.TypeMap())
```

Comment:
I believe Postgres has built-in support for UUID via the google/uuid package that we use throughout Dapr. I'd rather not add another dependency here if possible

Reply:
Currently there is an issue with pgx for uuid and numeric columns which returns uuid or numeric columns as a array of strings
[](https://github.com/dapr/components-contrib/issues/1016)

You can find the details about the fix here
[](https://github.com/jackc/pgx/wiki/UUID-Support)

Reply:
I would rather using something that is based on https://pkg.go.dev/github.com/google/uuid please - that's the package we use throughout Dapr